### DISCUSSION TITLE
A simple toggle button

### DISCUSSION THREAD

**Discussion by kookma** [NicolasPetton/Notebook#3] [Q&A] [answered]
Tiddlywiki 5.1.23 has a new filter operator causes use a simple code for switching colors:

```
<$button>Day/Night
<$action-listops $tiddler="$:/palette" $field="text" $subfilter="+[toggle[$:/palettes/Vanilla],[$:/palettes/SolarFlare]]" />
</$button>  
```

Give a try in https://tiddlywiki.com/prerelease

May be it can be used for switch palettes.

**Comment by kookma** [comment]
@NicolasPetton - Please ignore the above solution. One caveat is if user has set the $:/palette to something else (none of day/light palettes) the toggle just appends the value to the existing one. There is another solution but it is not simpler than the one you have used (e.g two $reveal widgets)

This alternative solution suggested by Saq in Google forum

```
<$button>Day/Night
<$action-listops $tiddler="$:/palette" $field="text" $filter="""[{$:/palette}!match[$:/palettes/Vanilla]!match[$:/palettes/SolarFlare]then[$:/palettes/Vanilla]] :else[enlist{$:/palette}toggle[$:/palettes/Vanilla],[$:/palettes/SolarFlare]]""" />
</$button>
```

**Reply by NicolasPetton** [reply]
Thanks (I saw the email thread as well ;-) )

**Comment by kookma** [comment, marked as answer]
@NicolasPetton - As you my follow my discussion with Saq on Google forum this is the most simple solution by now

```
<$vars
   lightPalette="$:/palettes/Vanilla"
   darkPalette="$:/palettes/SolarFlare"
>
<$button>Day/Night II
<$action-setfield $tiddler="$:/palette" text={{{[{$:/palette}match<lightPalette>then<darkPalette>else<lightPalette>]}}}/>
</$button>
</$vars>
```

You may read _darkPalette_ and _lightPalete_ from a config tiddler fields. This way you may let Notebook users to simply decide on their dark/light palettes instead of head coded them. I am not sure if it make sense to use other palettes or Notebook design restricts the palette to vanilla and beige.

**Reply by NicolasPetton** [reply]
Done in ab31d36477cfb19e85092a3e26b8814ab58f3a5e.
